### PR TITLE
feat : convert the admob quickstart to Kotlin

### LINKED ISSUE #622: [Internal] Translate admob app to kotlin
See #557

### PR DESCRIPTION
**Pull request by the-dagger**
@samtstern Please merge this after #641.
Fixes #622

### CODE REVIEW

File: admob/app/build.gradle [hunk 1]
```diff
@@ -43,6 +46,10 @@ dependencies {
     androidTestImplementation 'com.android.support.test.espresso:espresso-core:3.0.2'
     androidTestImplementation 'com.android.support.test:rules:1.0.2'
     androidTestImplementation 'com.android.support.test:runner:1.0.2'
+    compile "org.jetbrains.kotlin:kotlin-stdlib-jdk7:1.2.61"
```
Comment (samtstern) [inline admob/app/build.gradle]: compile --> implementation

File: admob/app/build.gradle [hunk 2]
```diff
@@ -43,6 +46,10 @@ dependencies {
...
 }
 
 apply plugin: 'com.google.gms.google-services'
+repositories {
```
Comment (samtstern) [inline admob/app/build.gradle]: I think this should be moved/removed

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+import com.google.android.gms.ads.AdRequest
+import com.google.android.gms.ads.InterstitialAd
+import com.google.samples.quickstart.admobexample.R
+import kotlinx.android.synthetic.main.activity_main.*
+
+
+class MainActivity : AppCompatActivity() {
+
+    private val TAG = "MainActivity"
+    // [END add_lifecycle_methods]
```
Comment (samtstern) [inline admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt]: Not sure what happened here but we have an "END" tag with no "START.

In Java the tag wraps the onPause, onResume, and onDestroy methods

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+    // [END_EXCLUDE]
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        val adRequest = AdRequest.Builder().build()
+
+        adView.loadAd(adRequest)
+        // [END load_banner_ad]
```
Comment (samtstern) [inline admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt]: Same here, missing `START load_banner_ad`

These comments are how snippets get into the docs.  So when translating from Java to Kotlin make sure that the START/END groups wrap code which has the same logical function.

File: admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+        if (!mInterstitialAd.isLoaded) {
+            requestNewInterstitial()
+        }
+    }
+
+    /** Called before the activity is destroyed  */
+    public override fun onDestroy() {
+        adView.destroy()
+        super.onDestroy()
+    }
```
Comment (samtstern) [inline admob/app/src/main/java/com/google/samples/quickstart/admobexample/kotlin/MainActivity.kt]: Looks like that lifecycle END tag from above just needs to be moved down here

File: admob/app/build.gradle (hunk 2 above)
Comment (the-dagger) [reply admob/app/build.gradle]: Not sure how this got in here :/

**Pull request merged**